apps/mobile-wallet/src/features/transactionsDisplay/TransactionModal.tsx

```diff
@@ -12,46 +21,135 @@ import AssetAmountWithLogo from '~/components/AssetAmountWithLogo'
+  const { t } = useTranslation()
+  const allAddressHashes = useUnsortedAddressesHashes()
+
+  const { data: tx, isLoading } = useFetchTransaction({ txHash })
+
+  if (isLoading)
+    return (
+      <EmptyPlaceholder>
+        <AppText size={32}>⏳</AppText>
+        <AppText>{t('Loading transaction details...')}</AppText>
+      </EmptyPlaceholder>
+    )
+
+  if (!tx) return null
+
+  const referenceAddress = findTransactionReferenceAddress(allAddressHashes, tx)
+
+  if (!referenceAddress) return null
+
+  return <TransactionDetailRows tx={tx} refAddressHash={referenceAddress} />
+}
+
+interface TransactionModalSubcomponentProps {
+  tx: e.AcceptedTransaction | e.PendingTransaction
+  refAddressHash: AddressHash
+}
+
+const TransactionDetailRows = ({ tx, refAddressHash }: TransactionModalSubcomponentProps) => {
+  const { t } = useTranslation()
+  const direction = useTransactionDirection(tx, refAddressHash)
+
   const isOut = direction === 'out'
-  const isMoved = infoType === 'move'
 
-  const openNftGridModal = () => dispatch(openModal({ name: 'NftGridModal', props: { nftsData } }))
+  return (
+    <>
+      <Row title={t('From')} transparent>
+        {isOut ? (
+          <AddressBadge addressHash={refAddressHash} />
+        ) : (
+          <IOList
+            currentAddress={refAddressHash}
+            isOut={isOut}
+            outputs={tx.outputs}
+            inputs={tx.inputs}
+            timestamp={isConfirmedTx(tx) ? tx.timestamp : undefined}
+          />
+        )}
+      </Row>
+      <Row title={t('To')} transparent>
+        {!isOut ? (
+          <AddressBadge addressHash={refAddressHash} />
+        ) : (
+          <IOList
+            currentAddress={refAddressHash}
+            isOut={isOut}
+            outputs={tx.outputs}
+            inputs={tx.inputs}
+            timestamp={isConfirmedTx(tx) ? tx.timestamp : undefined}
+          />
+        )}
+      </Row>
 
-  const groupedIOAmounts = groupBy(tokensWithSymbol, (t) => (t.amount > 0 ? 'in' : 'out'))
+      {isConfirmedTx(tx) && (
+        <Row title={t('Timestamp')} transparent>
+          <AppText semiBold>
+            {dayjs(tx.timestamp).toDate().toLocaleString(undefined, { dateStyle: 'medium', timeStyle: 'short' })}
+          </AppText>
+        </Row>
+      )}
+
+      <TransactionStatus tx={tx} />
+
+      <Row title={t('Fee')} transparent>
+        <Amount value={BigInt(tx.gasPrice) * BigInt(tx.gasAmount)} fullPrecision bold showOnDiscreetMode />
+      </Row>
+
+      <TransactionAmounts tx={tx} refAddressHash={refAddressHash} />
+    </>
+  )
+}
+
+const TransactionStatus = ({ tx }: Pick<TransactionModalSubcomponentProps, 'tx'>) => {
+  const { t } = useTranslation()
+  const theme = useTheme()
 
   const statuses = {
     confirmed: {
       color: theme.global.valid,
       text: t('Confirmed')
     },
     pending: {
-      color: theme.bg.primary,
+      color: theme.bg.contrast,
```

Comment:
This wasn't visible before because we didn't allow opening modals for pending txs. But I've witnessed many friends of mine that tried to click on a pending tx row and since we have now implemented it on desktop wallet as well, I implemented it on mw too.